main.py

```diff
@@ -109,6 +112,24 @@ def main(
     fmt = '%(asctime)s %(levelname)s (%(name)s %(lineno)s): %(message)s'
     coloredlogs.install(level='DEBUG' if verbose else 'INFO', fmt=fmt)
 
+    if workflow and workflow != 'list':
+        wfl_conf_path = to_path(__file__).parent / f'configs/defaults/{workflow}.toml'
+        assert wfl_conf_path.exists(), wfl_conf_path
+        wfl_conf_path = [str(wfl_conf_path)]
+    else:
+        wfl_conf_path = []
```

Comment:
It's not really part of this change, but I only just noticed this... Why is accepting `workflow=='list'` an option? There's a specific `--list-workflows` argument, so we're supporting this behaviour in two different ways?

Can we change the order of execution to:

1. load args
2. if list-workflows: do that, then quit
3. if list-last-stages: do that, then quit
4. all other behaviours require a valid workflow to be set

Reply:
Yes, it's ridiculous isn't it.

What I did was make the smallest change that doesn't change behaviour. One could make a slightly less small change that's more elegant, or remove the `list` special case.

:shrug: Probably just remove the special case. Surely no `--list-*` options are used in production, and I for one just look at the code in _main.py_ to see what's available, 'cos that's less painful than running it!

Reply:
I think the logic for list-workflows/list-endpoints is interesting, but it anticipates someone might pick up this codebase and use them to have a casual browse of the features. 

In reality the users of this pipeline are also developers of it, with a detailed knowledge of the workflows...

And also list-workflows also prints the last stages, so I'm not sure what utility having two options aimed at the same use-case was supposed to provide. I'd be OK with dropping both, but I'll make an issue for it and it can be a 10 second 'any of y'all use this option' discussion at the next prod-pipes meeting

Reply:
I think if you make an issue about whether we should have these options, that will already be 10x how much attention they are worth. IMHO we should just leave these options' half dozen lines as they are and forget about them again, as they're not hurting anything.

Reply:
Cleaned up the code a bit by removing the particularly ridiculous `list` special case.